google-cloud-storage/lib/google/cloud/storage/bucket.rb

```diff
@@ -757,6 +760,71 @@ def lock_retention_policy!
           true
         end
 
+        ##
+        # Whether the bucket's file IAM configuration enables Bucket Policy
+        # Only. The default is false. This value can be modified by calling
+        # {Bucket#policy_only=}.
+        #
+        # If true, access checks only use bucket-level IAM policies or above,
+        # all object ACLs within the bucket are no longer evaluated, and
+        # access-control is configured solely through the bucket's IAM policy.
+        # Any requests which attempt to use the ACL API to view or manipulate
+        # ACLs will fail with 400 errors.
+        #
+        # @return [Boolean] Returns `false` if the bucket has no IAM
+        #   configuration or if Bucket Policy Only is not enabled in the IAM
+        #   configuration. Returns `true` if Bucket Policy Only is enabled in
+        #   the IAM configuration.
+        #
+        # @example
+        #   require "google/cloud/storage"
+        #
+        #   storage = Google::Cloud::Storage.new
+        #
+        #   bucket = storage.bucket "my-bucket"
+        #
+        #   bucket.policy_only = true
+        #   bucket.policy_only? # true
```

Comment:
IAMConfiguration in the future is expected to have additional fields. Given the current design do you feel this is manageable?

Reply:
Flattening is well established in the design for configuration objects with a couple of fields (`Bucket#website_main=` and `Bucket#website_404` for example.) Do you anticipate a lot of complexity in IAMConfiguration?

Reply:
Based on prior conversations it should be similar fields that will be exposed here in the future.

Sgtm, I just wanted to double check. Thank you for the feedback.